lib/README.md

```diff
@@ -0,0 +1,30 @@
+# lowRISC Nix Libraries
+
+## The Documentation Library
+
+The `doc` library contains `buildMdbookSite`, which is a helper function for creating mdBook site builds.
+An example of it's use is below.
+
+```nix
+let
+  fs = pkgs.lib.fileset;
+in {
+  docs_build = doc.buildMdbookSite {
+    pname = "documentation";
+    version = "0.0.1";
+    nativeBuildInputs = [pkgs.python311];
```

Comment:
Just to make it super explicit, I would add a comment saying that adding to nativeBuildInputs would mostly be used for preprocessor deps.